### ISSUE TITLE
Create a Map (w/ Raylib features)  using a Chiventure context

### ISSUE THREAD

**Issue by damonhines** [uchicago-cs/chiventure#727]
This will involve creating a sample map using current chiventure structs and monkey-patching. This will provide a basis for how a map for next-year's class can be implemented via an Example. This will also be an example used for the demo as well

**Comment by damonhines** [comment]
Beginning:

 Upon coding a chiventure context, I noticed that raylib and ncurses cannot be used together. Both libraries alter a specific part of the computer and that causes conflict. Therefore, my map example must use room structs but not a chiventure context or mock in memory context game.

**Comment by damonhines** [comment]
Update:
I am successfully creating rooms and connections to create the map. The existing connection functions help layout the what rooms are in which direction. This will be helpful for the future graphics teams because they will better understand how the maps are set up from WDL files. I'm going to have a 5 way room connection where rooms 4 and 5 connect to room 3.

**Comment by damonhines** [comment]
Update:
I successfully created the map using chiventure logic and room connections. Now I am going to try and add a character to the map to indicate where the current player is. This will help show the benefit of having a map in chiventure so the player can see what room there is. In the future, the graphics team can research where to place this map in combination with the cli split screen

**Comment by damonhines** [comment]
Update:
I added the location of the player to the example using data from the game struct. In the future, the graphics team would need to make sure the location of the player is constantly updated. 

For now, i'm going to work on submitting a pull request.
![Screen Shot 2020-06-04 at 12 54 16 AM](https://user-images.githubusercontent.com/39503849/83720068-f1303a00-a5fd-11ea-8d0d-9f41c4c431df.png)

**Comment by damonhines** [comment]
Summary: 

The example has been merged to dev, and it consists of the 5-room map displayed above. This example uses the fact that rooms in Chiventure can be specified by cardinal coordinates, and this knowledge will come in handy when making a full version of the map. The main take away is that there needs to be an algorithm made to express/draw maps. I decided to explicitly make a T shaped map, but there are other ways to make the map shape. It may be good to review the current map code and learn how the map is generated. This is something to think about when coding a map in the future.
